dandi/girder.py

```diff
@@ -357,14 +357,20 @@ def get_download_file_iter(self, file_id, chunk_size=MAX_CHUNK_SIZE):
         """
         """
 
-        def downloader():
+        def downloader(start_at=0):
             # TODO: make it a common decorator here?
             # Will do 3 attempts to avoid some problems due to flaky/overloaded
             # connections, see https://github.com/dandi/dandi-cli/issues/87
             for attempt in range(3):
                 try:
-                    return self.downloadFileAsIterator(file_id, chunkSize=chunk_size)
-                    break
+                    path = f"file/{file_id}/download"
+                    params = None
+                    if start_at > 0:
+                        params = {"offset": str(start_at)}
```

Comment:
and coverage is right about this one... I added testing on draft version:
```
diff --git a/dandi/tests/test_download.py b/dandi/tests/test_download.py
index d3fcea3..4287d1b 100644
--- a/dandi/tests/test_download.py
+++ b/dandi/tests/test_download.py
@@ -146,12 +146,13 @@ def test_girder_tqdm(monkeypatch):
     with TQDMProgressReporter() as pr:
         pr.update(10)
 
+import itertools
 
-@pytest.mark.parametrize("resizer", [lambda sz: 0, lambda sz: sz // 2, lambda sz: sz])
-def test_download_000027_resume(tmp_path, resizer):
+@pytest.mark.parametrize("resizer,version", itertools.product([lambda sz: 0, lambda sz: sz // 2, lambda sz: sz], ["0.200721.2222", "draft"]))
+def test_download_000027_resume(tmp_path, resizer, version):
     from ..support.digests import Digester
 
-    url = "https://dandiarchive.org/dandiset/000027/0.200721.2222"
+    url = f"https://dandiarchive.org/dandiset/000027/{version}"
     digester = Digester()
     download(url, tmp_path, get_metadata=False)
     dsdir = tmp_path / "000027"
```

but test failed
```
dandi/tests/test_download.py:176: in test_download_000027_resume
    assert nwb.stat().st_size == size
E   assert 28188 == 18792
E     +28188
E     -18792
```
so I guess more work to be done for the girder downloader?

Reply:
It appears that gui.dandiarchive.org isn't honoring the `offset` parameter for the `file/{fileid}/download` endpoint, even though the Swagger UI documents the parameter, and even though I seem to recall it working when I was working on #189.  Switching to using a Range: header instead of `offset` works when testing with curl, but it doesn't work with the Girder client, as they still haven't merged https://github.com/girder/girder/pull/3301.

Reply:
Do you mean girder.Dandiarchive.org? There is nothing to talk about to gui.dandiarchive.org . How did get managed to resume in my example I pasted earlier?

Reply:
Yes, I meant girder.dandiarchive.org.  I'm guessing that `wget` used a Range: header, which it can do, because it's not written with the Girder client.

Reply:
So should we do the same directly? I think we know all metadata to compose the url or not? If not we could I guess issue head request to follow redirect and work with the end url directly

Reply:
It should be working now.